src/realtime/index.ts

```diff
@@ -0,0 +1,120 @@
+import { TigrisClientConfig } from "../tigris";
+// this needs to go in its own file
+import { EventEmitter } from "node:events";
+import { Transport } from "./transport";
+import { MessageEvent } from "./messages";
+
+type SubscribeCallback = (string) => void;
+
+export interface RealTimeConfig {
+	/*
+	 * An id used to identify this client when using the presence feature.
+	 * If the client id is not set, an error will be thrown when using any presence features
+	 */
+	clientId?: string;
+}
+
+export class RealTime {
+	private _config: RealTimeConfig;
+	private channelManager: ChannelManager;
+	private transport: Transport;
+	constructor(config: RealTimeConfig = {}) {
+		this.transport = new Transport();
+		this.channelManager = new ChannelManager(this.transport);
+
+		this._config = config;
+	}
+
+	async connect() {
+		await this.transport.isConnected();
+	}
+
+	getChannel(name: string): Channel {
+		return this.channelManager.getOrCreate(name);
+	}
+
+	close() {
+		this.channelManager.close();
+		this.transport.close();
+	}
+
+	socketId() {
+		return this.transport.socketId();
+	}
+}
+
+class Presence extends EventEmitter {
+	constructor() {
+		super();
+	}
+}
+
+export class Channel extends EventEmitter {
+	private name: string;
+	private hasAttached: boolean;
+	private transport: Transport;
+
+	constructor(name: string, transport: Transport) {
+		super();
+		this.hasAttached = false;
+		this.name = name;
+		this.transport = transport;
+	}
+
+	subscribe(msgName: string, cb: SubscribeCallback) {
```

Comment:
if we introduce change feed would it be a new API or some option in Channel or Subscribe API?

Reply:
I think it might be cleaner as a new API. Since it is a channel but not a channel that you would publish on it might be better as something like:

```
  let collectionFeed = realtime.getChangeFeed("todos");

```